src/app/event-external-root/event-external-root.controller.js

```diff
@@ -3,14 +3,34 @@
 angular.module('esn.calendar.event-consultation')
   .controller('CalEventExternalRootController', CalEventExternalRootController);
 
-function CalEventExternalRootController($scope, result) {
+function CalEventExternalRootController($rootScope, $location, result, calEventExternalRootService) {
   const self = this;
 
+  $rootScope.$on('UPDATE_ACTION_EXCAL', function(event, newLink) {
+    const newJWT = parseJwtKey(newLink);
+
+    if (newJWT) {
+      $location.search('jwt', newJWT);
+    }
+  });
+
   if (result.error) {
     self.error = result.error;
   } else {
     self.eventJcal = result.eventJSON;
     self.attendeeEmail = result.attendeeEmail;
-    self.links = result.links;
+    self.links = {};
+    Object.keys(result.links).forEach(action => {
+      const url = new URL(result.links[action]);
+      const jwt = url.searchParams.get('jwt');
+
+      self.links[action] = calEventExternalRootService.getChangeParticipationUrl(jwt);
+    });
+  }
+
+  function parseJwtKey(link) {
+    const token = link.match(/jwt=(.*)/);
+
+    return token ? token[1] : null;
```

Comment:
This is essentially doing the same thing as this:

```js
const url = new URL(result.links[action]);
const jwt = url.searchParams.get('jwt');
```

So you need to decide which one to use and put them in `parseJwt`.

Reply:
const url = new URL(result.links[action]); it is used for a valid URL and this is not the case for the link sent from calendar 
so we will use parseJwtKey